### ISSUE TITLE
Is it possible to indicate node rank?

### ISSUE THREAD

**Issue by lincolnfrias** [xflr6/graphviz#35]
I can't figure out to solve the problems posed by [this](http://stackoverflow.com/questions/34747843/graph-node-position-in-graphviz-in-python) and [this](http://stackoverflow.com/questions/40808940/how-to-align-nodes-in-graph-using-pythons-graphviz-module) stackoverflow's questions. Actually, I'm trying to reproduce [this](http://stackoverflow.com/questions/27504703/in-graphviz-how-do-i-align-an-edge-to-the-top-center-of-a-node/36953206#36953206) graph and can't figure out how to input the rank information on the python side. 

Any directions?

By the way, thanks for such a nice library!

**Comment by xflr6** [comment]
Thanks. At the moment, this is only possible by appending [custom DOT strings](https://graphviz.readthedocs.io/en/latest/manual.html#custom-dot-statements) to `g.body`.

I am thinking about a good API here, maybe a context manager (need to read up on the dot guide):

```python
with g.context(rank='same') as c:
   c.node('F', 'First state')
   c.node('I', '{<i1>In 1| <i2> In 2}', shape='record')
```

 Maybe you have ideas?

**Comment by lincolnfrias** [comment]
This a very good idea.
I'm new to graphviz, but I'll let you know if I came up with something :)

**Comment by xflr6** [comment]
Implemented in `0.7`:

```python
import graphviz

g = graphviz.Graph()
g.attr(splines='ortho')

with g.subgraph() as s:
    s.attr('node', width='0', shape='point', style='invis')
    for i in range(4):
        s.node('%d' % i)

with g.subgraph() as s:
    s.attr(rank='same')
    s.edges(['12', '23'])
	
g.edge('0', '2')
g.attr('node', shape='box')

with g.subgraph() as s:
    s.attr(rank='same')
    s.edges(['A0', '0B'])

g.edges(['1C', '1D', '3E', '3F'])

g.view()
```
![graph gv](https://cloud.githubusercontent.com/assets/6342379/25554196/68fd1b92-2cc7-11e7-8b3b-702c4423140b.png)

```python
>>> graphviz.__version__
'0.7'
```

**Comment by lincolnfrias** [comment]
Awesome @xflr6. Thank you very much for solving this.
